modules/images/webp-uploads/load.php

```diff
@@ -111,6 +111,102 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
+ *
+ * @since n.e.x.t
+ *
+ * @param array $metadata The current metadta of the modified image.
+ * @param int   $attachment_id The ID of the attachment being modified.
+ * @return array The metadata modified of the image.
+ */
+function webp_uploads_create_sources_property_for_full_image( $metadata, $attachment_id ) {
+	// This should take place only on the JPEG image.
+	$valid_mime_transforms = webp_uploads_get_supported_image_mime_transforms();
+
+	// Not a supported mime type to create the sources property.
+	$mime_type = get_post_mime_type( $attachment_id );
+	if ( ! isset( $valid_mime_transforms[ $mime_type ] ) ) {
+		return $metadata;
+	}
+
+	$file = get_attached_file( $attachment_id, true );
+
+	// File does not exist.
+	if ( ! file_exists( $file ) ) {
+		return $metadata;
+	}
+
+	if ( ! isset( $metadata['width'], $metadata['height'] ) ) {
+		return $metadata;
+	}
+
+	if ( ! isset( $metadata['sources'] ) || ! is_array( $metadata['sources'] ) ) {
+		$metadata['sources'] = array();
+	}
+
+	if ( empty( $metadata['sources'][ $mime_type ] ) ) {
+		$metadata['sources'][ $mime_type ] = array(
+			'file'     => wp_basename( $file ),
+			'filesize' => filesize( $file ),
+		);
+		wp_update_attachment_metadata( $attachment_id, $metadata );
+	}
+
+	$allowed_mimes      = array_flip( wp_get_mime_types() );
+	$original_directory = pathinfo( $file, PATHINFO_DIRNAME );
+	$original_extension = pathinfo( $file, PATHINFO_EXTENSION );
+
+	foreach ( $valid_mime_transforms[ $mime_type ] as $targeted_mime ) {
+		// If this property exists no need to create the image again.
+		if ( ! empty( $metadata['sources'][ $targeted_mime ] ) ) {
+			continue;
+		}
+
+		if ( ! isset( $allowed_mimes[ $targeted_mime ] ) || ! is_string( $allowed_mimes[ $targeted_mime ] ) ) {
+			continue;
+		}
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => $targeted_mime ) ) ) {
+			continue;
+		}
+
+		$editor = wp_get_image_editor( $file );
+
+		if ( is_wp_error( $editor ) ) {
+			continue;
+		}
+
+		$editor->resize( $metadata['width'], $metadata['height'], false );
+		$extension = explode( '|', $allowed_mimes[ $targeted_mime ] );
+		$extension = reset( $extension );
+
+		if ( empty( $extension ) || empty( $original_extension ) ) {
+			continue;
+		}
+
+		$filename         = wp_basename( $file, ".{$original_extension}" );
+		$target_file_name = trailingslashit( $original_directory ) . "{$filename}.{$extension}";
+		$image            = $editor->save( $target_file_name, $targeted_mime );
+
+		if ( is_wp_error( $image ) ) {
+			continue;
+		}
+
+		$metadata['sources'][ $targeted_mime ] = array(
+			'file'     => $image['file'],
+			'filesize' => isset( $image['path'] ) ? filesize( $image['path'] ) : 0,
+		);
+
+		wp_update_attachment_metadata( $attachment_id, $metadata );
+	}
+
+	return $metadata;
+}
+
+add_filter( 'wp_generate_attachment_metadata', 'webp_uploads_create_sources_property_for_full_image', 10, 2 );
```

Comment:
Can we add to our existing callback rather than hooking again? that way our fix for constrained systems will work with the full size as well.

Reply:
Sure, just wanted the avoid a large function but I see the benefit in keeping everything in a single place.